dsc_lib/src/dscresources/dscresource.rs

```diff
@@ -50,6 +50,8 @@ pub enum Capability {
     Set,
     /// The resource supports the `_exist` property directly.
     SetHandlesExist,
+    /// The resource supports the `what-if` execution type directly.
+    SetHandlesWhatIf,
```

Comment:
This shouldn't be specific to `set` although primarily used by `set`.  `get` can also fail if the user doesn't have the right permission.  Should generalize as just `WhatIf`.

Reply:
So the expectation is if the resource implements `what-if` directly - it's for every command? 

Should I add `get` support for `what-if` separately? I haven't added `what-if` to any commands besides `config set` so the `get` scenario won't work yet.

Reply:
No, a resource can implement it for whatever operation makes sense.  `set` would be recommended, but it could apply to `get`, `test`, and even `export`.

Reply:
-whatif is not a common parameter for get-* cmdlets, by comparison.